homeassistant/components/ssdp/__init__.py

```diff
@@ -401,11 +399,8 @@ async def async_scan_broadcast(self, *_: Any) -> None:
         # address. This matches pysonos' behavior
         # https://github.com/amelchio/pysonos/blob/d4329b4abb657d106394ae69357805269708c996/pysonos/discovery.py#L120
         for listener in self._ssdp_listeners:
-            try:
-                IPv4Address(listener.source_ip)
-            except ValueError:
-                continue
-            await listener.async_search((str(IPV4_BROADCAST), SSDP_PORT))
+            if len(listener.source) == 2:
```

Comment:
Maybe make this `_is_listener_source_ipv4` so it shows intent?

```suggestion
            if _is_listener_source_ipv4(listener.source):
```

Reply:
Yes, good point. Actually, I'll add this to async-upnp-client but lets wait until this PR is all tested/working before I release `async-upnp-client==0.24.1`?